mlflow/utils/environment.py

```diff
@@ -19,16 +19,23 @@ def _mlflow_conda_env(path=None, additional_conda_deps=None, additional_pip_deps
                  in dictionary format.
     :param additional_conda_deps: List of additional conda dependencies passed as strings.
     :param additional_pip_deps: List of additional pip dependencies passed as strings.
-    :param additional_channels: List of additional conda channels to search when resolving packages.
+    :param additional_conda_channels: List of additional conda channels to search when resolving
+                                      packages.
     :return: ``None`` if ``path`` is specified. Otherwise, the a dictionary representation of the
              Conda environment.
     """
-    env = yaml.safe_load(_conda_header)
-    env["dependencies"] = ["python={}".format(PYTHON_VERSION)]
     pip_deps = (["mlflow"] if install_mlflow else []) + (
         additional_pip_deps if additional_pip_deps else [])
-    if additional_conda_deps is not None:
-        env["dependencies"] += additional_conda_deps
+
+    pip_not_specified = (additional_conda_deps is None
+                         or not [i for i in additional_conda_deps if "pip==" in i or "pip" == i])
```

Comment:
@tfurmston I'm not sure that this logic accounts for the full suite of possible version specification formats used in a conda env file (e.g., `pip>=`, `pip<=`, `pip20.0.02|20.0.2*`). Other examples for `numpy` are listed here (these apply to all packages): https://docs.conda.io/projects/conda-build/en/latest/resources/package-spec.html

Given that it's difficult to accurately determine whether `pip` is already included in a given environment, I think it would be safer to either:

1. Revert changes to this method and update each of the module-specific `DEFAULT_CONDA_ENV`s to include the `pip` dependency

or

2. Always add an unversioned `pip` dependency to the conda environment, regardless of whether or not `pip` is already specified in the list of `additional_pip_deps` passed to this method. In fact, we could update the docs to explicitly indicate that `pip` is automatically added as a dependency if `additional_pip_deps` is non-empty and therefore should not be specified within `additional_pip_deps`. 

I'm in favor of option 2. cc'ing @tomasatdatabricks for an opinion here.

Reply:
Um, that is a good point. I had not considered those cases. 

Yeah, in that case it does seem like it would be brittle to handle all of these different cases. 

Personally, I would also prefer option 2, but am very happy to do whatever you and @tomasatdatabricks think is best. :)

Reply:
I suppose an alternative would be to follow the approach from conda for this type of check?

See here, [https://github.com/conda/conda/blob/master/conda_env/env.py#L54](https://github.com/conda/conda/blob/master/conda_env/env.py#L54)

What do you think?

Reply:
I also like option 2. Adding dependency on pip without version is safe even if pip is already included in dependencies with or without version.

The conda code looks ok but I don't think we need it, better to keep this simple.

Reply:
ok, no problem. I've made the change. 

My last point was was about why this functionality in conda is not actually being called. Part of that functionality seems to address the same point as this MR, but obviously for conda in general. 

Maybe I am overthinking it, but I just thought it was worth highlighting.

Reply:
Ah I see, yeah that is a good question. Maybe you can file a ticket with conda as well. 

But I think adding unversioned pip is a good idea regardless. At the very least it will get rid of the warning.

Reply:
ok, cool. That makes sense. 

Thanks